uber/models.py

```diff
@@ -1243,9 +1196,10 @@ def _badge_adjustments(self):
 
         self.badge_type = get_real_badge_type(self.badge_type)
 
+        if self.orig_value_of('badge_type') != self.badge_type:
+            self.session.update_badge(self, self.orig_value_of('badge_type'), self.orig_value_of('badge_num'))
```

Comment:
So this is basically saying if the `badge_type` changed, always update the badge? Thus resolving the issue outlaw was having where changing the badge type would throw an error?